xcdat/regridder/regrid2.py

```diff
@@ -0,0 +1,547 @@
+
+        weight = np.sin(np.deg2rad(north_bounds)) - np.sin(np.deg2rad(south_bounds))
+
+        weights.append(weight.values.reshape(contrib.shape[0], 1))
+
+    return mapping, weights
+
+
+def pertub(value):
+    """
+    Pertub a value.
+
+    Parameters
+    ----------
+    value :
+        Value to pertub.
+    """
+    if value >= 0.0:
+        offset = np.ceil(value + 0.000001)
+    else:
+        offset = np.floor(value - 0.000001) + 1.0
+
+    return offset
+
+
+# vectorize version of pertub
+vpertub = np.vectorize(pertub)
+
+
+def align_axis(
+    src_west: xr.DataArray, src_east: xr.DataArray, dst_west: xr.DataArray
+) -> Tuple[xr.DataArray, xr.DataArray, int]:
+    """
+    Aligns a longitudinal source axis to the destination axis.
+
+    Parameters
+    ----------
+    src_west : xr.DataArray
+        DataArray containing the western source bounds.
+    src_east : xr.DataArray
+        DataArray containing the eastern source bounds.
+    dst_west : xr.DataArray
+        DataArray containing the western destination bounds.
+
+    Returns
+    -------
+    Tuple[xr.DataArray, xr.DataArray, int]
+        A tuple containing the shifted western source bounds, the shifted eastern
+        source bounds, and the number of places shifted to align axis.
+    """
+    west_most = np.minimum(dst_west[0], dst_west[-1])
+
+    alignment_index = pertub((west_most - src_west[-1]) / 360.0).values
+
+    if src_west[0] < src_west[-1]:
+        alignment_index += 1
+    else:
+        alignment_index -= 1
+
+    src_alignment_index = np.where(
+        vpertub((west_most - src_west) / 360.0) != alignment_index
+    )[0][0]
+
+    if src_west[0] < src_west[-1]:
+        if west_most == src_west[src_alignment_index]:
+            shift = src_alignment_index
+        else:
+            shift = src_alignment_index - 1
+
+            if shift < 0:
+                shift = src_west.shape[0] - 1
+    else:
+        shift = src_alignment_index
+
+    src_length = src_west.shape[0]
+
+    shifted_indexes = np.arange(src_length + 1) + shift
+
+    wrapped = np.where(shifted_indexes > src_length - 1)
+
+    shifted_indexes[wrapped] -= src_length
+
+    shifted_src_west = src_west[shifted_indexes] + 360.0 * vpertub(
+        (west_most - src_west[shifted_indexes]) / 360.0
+    )
+
+    shifted_src_east = src_east[shifted_indexes] + 360.0 * vpertub(
+        (west_most - src_west[shifted_indexes]) / 360.0
+    )
+
+    if src_west[-1] > src_west[0]:
+        if shifted_src_west[0] > west_most:
+            shifted_src_west[0] += -360.0
+            shifted_src_east[0] += -360.0
+    else:
+        if shifted_src_west[-1] > west_most:
+            shifted_src_west[-1] += -360.0
+            shifted_src_east[-1] += -360.0
+
+    return shifted_src_west, shifted_src_east, shift
+
+
+def map_longitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
+    """
+    Map source to destination longitude.
```

Comment:
```suggestion
    """Map source to destination longitude.
```
Add description in docstring on how the algorithm/logic works